scalgrafanalib/core.py

```diff
@@ -0,0 +1,71 @@
+from typing import Any, Dict, TypeVar
+import attr
+from grafanalib import core  # type: ignore
+
+Json = Dict[str, Any]
+Self = TypeVar("Self")
+
+
+@attr.s
+class Tooltip(core.Tooltip):
+    """SimpleTooltip : simple, "modern" tooltip configuration
+    Inherit from Tooltip to allow using in place of "usual" tooltip class"""
+
+    show: bool = attr.ib(default=True, validator=attr.validators.instance_of(bool))
+    showHistogram: bool = attr.ib(  # pylint: disable=invalid-name
+        default=True, validator=attr.validators.instance_of(bool)
+    )
+
+    def to_json_data(self) -> Json:
+        return {"show": self.show, "showHistogram": self.showHistogram}
+
+
+@attr.s
+class Target(core.Target):
+    """Target: set default `intervalFactor` mode to 1"""
+
+    def to_json_data(self) -> Json:
+        self.intervalFactor = 1
+        return super().to_json_data()
+
+
+@attr.s
+class TimeSeries(core.TimeSeries):
+    """TimeSeries: Allow settings decimals"""
+
+    decimals: int = attr.ib(default=0, validator=attr.validators.instance_of(int))
+
+    def to_json_data(self) -> Json:
+        json = super().to_json_data()
+        if self.decimals:
+            json["options"]["decimals"] = self.decimals
+        return json
+
+
+def _simplify(json: Json) -> Json:
+    return dict(filter(lambda item: item[1] is not None, json.items()))
```

Comment:
pythonic way: `{ key: value for key, value in json.items() if value is not None }